asmcomp/arm/emit.mlp

```diff
@@ -796,30 +801,48 @@ let emit_instr i =
           `	pop	\{trap_ptr, pc}\n`; 2
         end
 
+(* Upper bound on the size of the code sequence for a Linear instruction,
+   in 32-bit words. *)
+
+let max_instruction_size i =
+  match i.desc with
+  | Lswitch jumptbl ->
+      if !arch > ARMv6 && !thumb
+      then 1 + (Array.length jumptbl + 1) / 2 + Array.length jumptbl
+      else 2 + Array.length jumptbl
+  | _ ->
+      4   (* conservative upper bound *)
```

Comment:
This number doesn't appear to be conservative, e.g. for `Iextcall`.  Could you double-check all the cases?
Maybe there should be an assertion in `emit_all` that `emit_instr` doesn't return a number greater than what `max_instruction_size` returned.

(Aside: problems like this make me increasingly think that the right way to do code emitters is via a DSL, where absolutely everything including layout is described; then it's easier to do things such as measuring.)